modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/servlet/taglib/BaseDisplayTag.java

```diff
@@ -118,6 +118,8 @@ protected Map<String, Object> prepareProps(Map<String, Object> props) {
 			props.put("additionalProps", _additionalProps);
 		}
 
+		props.put("namespace", _randomNamespace);
+
```

Comment:
While this is ok, we can do better!

We can set namespace to be portlet namespace by default. See [how we are doing this](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/internal/servlet/taglib/BaseContainerTag.java#L142-L158) in base tag class in `frontend-taglib-clay`.

In 99% of the cases, this is the namespace devs need. We need portlet namespace to prefix portlet `actionsURL`s, `renderURL`s, `<portlet:namespace />`'d DOM elements etc. It is still good to have `namespace` attribute as a part of tag API, for some edge cases like portlets importing JSP directly from another module.